### PR TITLE
feat: add .mxf file support

### PR DESCRIPTION
**Pull request by timonrieger**
## Description

Adds support for MXF (Material Exchange Format) video files in Immich.

**What was added:**
- Added `.mxf` extension to the supported video formats list
- Registered `application/mxf` as the mime type for MXF files (official SMPTE mime type)
- Updated asset type classification to correctly identify MXF files as video assets
- Updated tests to validate MXF file handling

**Technical details:**
MXF is a container format with the mime type `application/mxf` (not `video/mxf`). The `assetType()` function was updated to check file extensions first (against the `video` and `image` objects) before falling back to mime type prefixes. This ensures container formats like MXF are correctly classified based on their extension listing, regardless of mime type prefix.

**Rationale for 5316bcf:**
Container formats (MXF, MOV, MP4, MKV, WebM) can have non-standard mime types. File extensions are more reliable for classification since Immich maintains explicit extension lists for video/image types. By checking extensions first, we ensure that any file extension listed in the `video` object is correctly classified as `AssetType.Video`, enabling proper playback, thumbnail generation, and metadata extraction.

Fixes #24640

## How Has This Been Tested?

- [x] Added test case for MXF file extension to mime type mapping
- [x] Updated test to allow `application/mxf` in video mime types validation
- [x] Verified MXF files are correctly classified as `AssetType.Video` during upload
- [x] Confirmed video playback endpoint works for MXF files (`/api/assets/{id}/video/playback`)
- [x] Verified thumbnail generation works for MXF files
- [x] Tested that existing video formats (MP4, MOV, etc.) continue to work correctly

Sample files downloaded from https://toolsfairy.com/video-test/sample-mxf-files were used to test the implementation.

EDIT: tested this now with two big video .mxf files i got sent from a fellow Discord user. See screenshots and recordings below

<details><summary><h2>Screenshots</h2></summary>

*Note: the last file upload fails due to size constraints of github codespaces (also happens on standard .mov files)*

https://github.com/user-attachments/assets/3e3c895c-cb4b-4f3d-a9eb-299e3130d6b7

EDIT (after test with "real" images/videos):

_Selecting .mxf files allowed now_

image

_Upload (20x, slow machine 🤣 )_

https://github.com/user-attachments/assets/b6c2cc52-7229-4fec-9c3d-4c63045965a9

_Replay (2x)_

https://github.com/user-attachments/assets/9ef40472-081a-4241-8e2c-f0410c6a4509

</details>

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [x] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [x] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [x] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

## Please describe to which degree, if any, an LLM was used in creating this pull request.

Used LLM assistance to research MXF mime type specifications and identify the classification logic issue.

### CODE REVIEW

**Comment by timonrieger** [comment]
~~agreed in discord to test not only against sample files, but some "real" files considering .mxf being a broad container format. Will get there on the weekend, due to file size limitations for Github Codespaces.~~ done, see updated PR description

**Comment by timonrieger** [comment]
the only thing behaving unexpectedly was that on upload finish the UI showed "Upload failed", however the file was visible and uploaded correctly. Maybe unrelated to the .mxf support addition.

File: server/src/utils/mime-types.ts [hunk 1]
```diff
@@ -133,6 +134,14 @@ export const mimeTypes = {
...
   toExtension,
   assetType: (filename: string) => {
+    // Check file extension first to handle cases like MXF (application/mxf) that are video containers
+    if (isType(filename, video)) {
+      return AssetType.Video;
+    }
+    if (isType(filename, image)) {
+      return AssetType.Image;
+    }
+    // Fallback to mime type check for any edge cases
```
Comment (timonrieger) [inline server/src/utils/mime-types.ts]: all tests are passing and this should be safe since all formats above are explicitly set, however to be safest we could use a single if branch for the mxf extension and revert these additions

Comment (jrasm91) [reply server/src/utils/mime-types.ts]: I think I'd probably prefer to remove these and add an additional `else if contentType === 'application/mxf`)` statement below.

**Review by jrasm91 — APPROVED** [review]
